test-case/check-volume-levels.sh

```diff
@@ -0,0 +1,341 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+set -e
+
+##
+## Case Name: check-volume-levels
+## Preconditions:
+##    topology is nocodec with loopack in PCM0P -> PCM0C
+##    aplay should work
+##    arecord should work
+##    PCM0C capture PGA supports -50 - +30 dB range
+##    PCM0C capture PGA supports mute switch
+## Description:
+##    Set volume and mute switch to various values, measure
+##    volume gain from the actual levels and compare to set gains.
+## Case step:
+##    1. Start aplay
+##    2a. Capture 1st wav file
+##    2b. Capture 2nd wav file
+##    2c. Capture 3rd wav file
+##    3. Measure volume gains
+## Expect result:
+##    command line check with $? without error
+##
+
+TESTDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)
+
+# shellcheck source=case-lib/lib.sh
+source "$TESTDIR/case-lib/lib.sh"
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $''TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+func_opt_parse_option "$@"
+
+tplg=${OPT_VALUE_lst['t']}
+
+TPLGREADER="$TESTDIR"/tools/sof-tplgreader.py
+PCM_ID=0
+CAP_FORMAT="S16_LE"
+CAP_RATE="48000"
+VOL_P30DB=80
+VOL_P10DB=60
+VOL_0DB=50
+VOL_M10DB=40
+VOL_M20DB=30
+VOL_M30DB=20
+VOL_MAX=$VOL_P30DB
+VOL_MIN=1
+VOL_MUTE=0
+VOL_NOM=$VOL_0DB
+APLAY_WAV=$(mktemp --suffix=.wav)
+ARECORD_WAV1=$(mktemp --suffix=.wav)
+ARECORD_WAV2=$(mktemp --suffix=.wav)
+ARECORD_WAV3=$(mktemp --suffix=.wav)
+
+#
+# Main test procedure
+#
+
+main () {
+    # Preparations
+    do_preparations
+    generate_sine
+
+    # Start sine playback
+    aplay_helper "$APLAY_WAV" & aplayPID=$!
+    pause
+
+    # Do capture tests
+    run_test_1
+    run_test_2
+    run_test_3
+
+    # Stop sine playback and do cleanup
+    pause
+    dlogi "The test procedure is now complete. Killing the aplay process $aplayPID."
+    kill $aplayPID
```

Comment:
This doesn't work for me sometimes, that's why I suggest to replace it with 'killall aplay'

Reply:
I didn't want use it if there would another test running with different sound device. What kind of problem did you see?

Reply:
If this fails only "sometimes" then it sounds like this test found a real audio bug . @keyonjie can you please elaborate?

Reply:
it reports error sometimes like this:
```
2020-10-19 05:39:45 UTC [INFO] The test procedure is now complete. Killing the aplay process 9360.
./check-volume-levels.sh: line 81: kill: (9360) - No such process
```

Reply:
If the recording process is not severely delayed it sounds like a bug. The playback sine wave duration is 60s and for me the test completes in less than 30s so there should be nearly 30s margin. Could the sample rates be severely off in the platform where you test (e.g. 10s recording takes much more time)?

Reply:
> ./check-volume-levels.sh: line 81: kill: (9360) - No such process

If the process is already dead then `killall` would: 1. hide what looks like a bug 2. possibly kill other, unrelated processes.

Reply:
@singalsu @marc-hb OK let's keep it as what it is today in the PR, as it doesn't introduce any real issue except seeing this warning, we can fix it later.